### PR TITLE
19154 Update TING validation rules

### PR DESCRIPTION
**Pull request by chenhongjing**
*Issue #:* /bcgov/entity#19154

*Description of changes:*
- check for duplicate TINGs
- check for at least 2 TINGs
- unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/filings/validations/amalgamation_application.py [hunk 1]
```diff
@@ -163,6 +172,20 @@ def validate_amalgamating_businesses(  # pylint: disable=too-many-branches,too-m
                         'path': amalgamating_businesses_path
                     })
 
+    if duplicate_businesses:
+        error_msg = 'Duplicate amalgamating business entry found in list: ' + \
```
Comment (chenhongjing) [inline legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:176]: I changed error messages returned after asking Yui.

File: legal-api/src/legal_api/services/filings/validations/amalgamation_application.py [hunk 2]
```diff
@@ -163,6 +172,20 @@ def validate_amalgamating_businesses(  # pylint: disable=too-many-branches,too-m
...
+            ', '.join(duplicate_businesses) + '.'
+        msg.append({
+            'error': error_msg,
+            'path': amalgamating_businesses_path
+        })
+
+    if len(amalgamating_businesses) < 2:
+        msg.append({
+            'error': 'Two or more amalgamating businesses required.',
```
Comment (chenhongjing) [inline legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:185]: Same as the above.

File: legal-api/src/legal_api/services/filings/validations/amalgamation_application.py [hunk 3]
```diff
@@ -98,9 +101,15 @@ def validate_amalgamating_businesses(  # pylint: disable=too-many-branches,too-m
                 is_any_ccc = True
             elif business.legal_type == Business.LegalTypes.BC_ULC_COMPANY.value:
                 is_any_ulc = True
-        elif ((corp_number := amalgamating_business_json.get('corpNumber')) and corp_number.startswith('A')):
-            if ((foreign_jurisdiction := amalgamating_business_json.get('foreignJurisdiction')) and
-                    foreign_jurisdiction.get('country') == 'CA' and foreign_jurisdiction.get('region') == 'BC'):
+        elif corp_number := amalgamating_business_json.get('corpNumber'):
+            if corp_number in amalgamating_businesses:
+                duplicate_businesses.append(corp_number)
+            amalgamating_businesses[corp_number] = amalgamating_business_json
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:107]: `amalgamating_businesses` list is used to hold `Business` object (ie. a businesses which are in lear db). if you include foreign business into it then the below code fails to (`if amalgamating_business:`) identify the business

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/amalgamation_application.py:107]: just realized we get the amalgamating_business using identifier below. it should work

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by argush3 — APPROVED** [review]

**Pull request merged**